waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -246,6 +246,18 @@
                  :waiter/pod-expired pod-expired})
       true)))
 
+(defn generate-port-with-offset
```

Comment:
We should write this function in a manner that ensures we never return a value lower than `pod-base-port`.